src/api/procedures/claimDividends.ts

```diff
@@ -24,44 +19,39 @@ export async function prepareClaimDividends(
 ): Promise<void> {
   const {
     context: {
-      polymeshApi: { tx, query },
+      polymeshApi: { tx },
     },
     context,
   } = this;
   const {
-    distribution: {
-      targets: { identities, treatment },
-      id: localId,
-      ticker,
-      paymentDate,
-      expiryDate,
-    },
+    distribution,
+    distribution: { id: localId, ticker, paymentDate, expiryDate },
   } = args;
 
   assertDistributionOpen(paymentDate, expiryDate);
 
-  const { did: currentDid } = await context.getCurrentIdentity();
+  const identity = await context.getCurrentIdentity();
 
-  const found = !!identities.find(({ did }) => did === currentDid);
+  const participant = await distribution.getParticipant({ identity });
```

Comment:
Fetching the current Identity will be unnecessary when `getParticipant` is modified to have optional arguments